app/src/main/java/org/gdg/frisbee/android/pulse/PulseFragment.java

```diff
@@ -76,13 +77,21 @@ public void onAttach(final Activity activity) {
     }
 
     @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putIntArray(INSTANCE_STATE_POSITIONS, mAdapter.getPositions());
+    }
+
+    @Override
     public void onActivityCreated(Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
 
         mTarget = getArguments().getString(ARG_TARGET);
         mMode = getArguments().getInt(ARG_MODE);
 
-        mAdapter = new PulseAdapter(getActivity());
+        int[] positions = savedInstanceState != null
+                ? savedInstanceState.getIntArray(INSTANCE_STATE_POSITIONS) : null;
+        mAdapter = new PulseAdapter(getActivity(), positions);
```

Comment:
By meaning positions, I mean that rankings of the countries/cities. The ones related to the numbering problem in the screenshot in the original issue.